File: data-structure-and-algorithm/leetcode/0014.Longest-Common-Prefix/LongestCommonPrefix.go
package main

import "fmt"

func longestCommonPrefix(strs []string) string {
	if len(strs) == 0 {
		return ""
	}
	if len(strs) == 1 {
		return strs[0]
	}
	first := strs[0]
	for maxLen := 0; maxLen < len(first); maxLen++ {
		for index := 1; index < len(strs); index++ {
			if maxLen >= len(strs[index]) || strs[index][maxLen] != first[maxLen] {
				return first[0:maxLen]
			}
		}
	}
	return first
}

func main() {
	fmt.Print(longestCommonPrefix([]string{"flower", "flow", "flight"}))
}

File: data-structure-and-algorithm/leetcode/0014.Longest-Common-Prefix/LongestCommonPrefix.java
public class LongestCommonPrefix {
    public String longestCommonPrefix(String[] strs) {
        if(strs.length == 0){
            return "";
        }
        if(strs.length == 1){
            return strs[0];
        }
        String first = strs[0];
        int len = 0;
        for(int i = 0; i < first.length(); i++){
            for(int j = 1; j < strs.length; j++){
                if(strs[j].length() <= i || strs[j].charAt(i) != first.charAt(i)){
                    return first.substring(0, i);
                }
            }
        }
        return first;
    }
}
